web-console/knikubevirt/Affinity/affinity.md

```diff
@@ -0,0 +1,61 @@
+# Affinity
+
+### About Affinity
+Affinity allows the user to create a relationship between two resources when scheduling them on the cluster.
+There are two high level relationship types:
+Affinity - attaching a group of resources together, making them run together in the same environment.
+Anti-affinity - creating a rejection relationship between a group of resources, thus avoiding scheduling any of them in the same environment.
+
+Besides defining if we want to attach or reject resources to one another, we can choose to "force" this rule we are creating.
+"Forcing" a rule means that the conditions of this rule must met, even if it means that some resources just won't be scheduled at all.
+
+### a few techincal words
+Affinity rules are Pod Properties, written in the pod deployment file. When a pod is scheduled, the deployment file states that there is an affinity rule condition which needs to be met.
+Applying an affinity rule to a resources is done via labels. Objects in OpenShift are ephemeral, and the way to target the same object which keeps crashing and regenerating is by its label.
+
+
+## Affinity rules - list page
+
+![Affinity list](img/D2-0-0.jpg)
+All affinity rules within the system can be viewed and managed from this page. It is located in the main navigation, under 'Workloads' 
+
+## Creating an Affinity rule from Affinity Rules page
+
+Clicking on the "Add Affinity rule" button at the top of the affinity list will open up the "Add Affinty rule" modal.
+
+![Affinity rule modal - default](img/D2-1-0.jpg)
+Going top to bottom, the user will choose to create an affinity (attachment), which is pre-selected, or change to Anti-affinity (rejection).
+Then, the user will select the resources he wishes to apply that rule to by adding their representative labels to the "Select resources by label" field.
+
+
+![Force rule](img/D2-1-1.jpg)
```

Comment:
I think the resources would be chips right? https://www.patternfly.org/v4/documentation/react/components/chipgroup/

Reply:
Not sure I follow.
As I understand, Chip is a visual represemntation of "applied filers".
Any entity with the following tag/attribute will be displayed via the chip's counter, followed by some list display method. It does not point to a specific entity. It does nothing by itself, only filters the "following". 
If we are talking about how to display the group of resources, then we have the labels for that.

Reply:
ya that's true, my main point here is that we don't have the types of labels in PF4 and I don't think we should design new ones. So pulling from PF4 chips would be the closest thing, otherwise we should propose that labels have the ability to add an icon.

Reply:
I agree that it is problematic to place Affinity rules under the Workloads tab. While it may be the most intuitive place to find it, it is not a workload per se.
I'm offering two alternatives in "Affinity rule list placement" on this doc:
https://docs.google.com/document/d/10l_4BQ-Wyx_n87y9-bzNScEtSz0ycDM3OxSr4kuK_bM/edit#heading=h.askclzqng25z

Summary:
These Affinity rules have two flows, from a specific workload (K8), and one as an abstract rule applied per label (Virtualization logic applied through k8 mechanism). Because of that, we need access to a list abstracted from workloads with an action to create a new (abstract) rule.
"Home" tab for Administrator persona and "Advanced" tab for Developer persona (toggled at the top of the side nav). But that consistency is not that important in terms of use-case. But It is somewhat not that important as the rest of the items there (I suspect).
"Administration" tab for Administrator persona is also an option since affinity by labels (Affinity groups) is also a sort of "Setting" and it is less prominent than the "home" tab

Reply:
I've followed up on this with Matt Carrano on the Patternfly side and it looks like there is some ongoing discussion around tags.
https://github.com/patternfly/patternfly-next/issues/759
I think we can hold on updating until things are more solid on the Patternfly side.